### ISSUE TITLE
Drag and drop table rows

### ISSUE THREAD

**Issue by alainacoppa** [rackerlabs/canon#144]
CDN has a need for drag and drop table rows and I believe @annwa also needs it for MyRack.  I have a design and desired behavior below.

How many CPs would want to use this before it would be considered as a Canon pattern? If we don't want it as a Canon pattern now, is there anything that I should try to communicate to the Backups team in terms of how it's built so that it can be more easily incorporated later?

![](http://4171e85bd5350b7f3558-7c534b3f9f8fab11de918744d806e281.r77.cf5.rackcdn.com/TinyGrab%20Screen%20Shot%202-9-15,%202.23.46%20PM.png)

Similar behavior to this: http://jqueryui.com/sortable/#placeholder

@annwa also interested in any design feedback.

**Comment by bradgignac** [comment]
Seems like this is fine to go in. From a Canon perspective, it looks like this is mostly a new icon.

**Comment by bradgignac** [comment]
It would be worth knowing how this impacts accessibility before making this a recommendation.

**Comment by alainacoppa** [comment]
@bradgignac agreed about the accessibility. @annwa do you think it would be enough to just add keyboard support? Similar to: http://test.cita.illinois.edu/aria/draganddrop/draganddrop1.php
Or do you think there needs to be full alternative way of moving items?

**Comment by annwa** [comment]
so that example is really difficult. i had a really hard time moving things around and if you had to guess how it operated without any visual cues i think you'd be really lost

i'm not 100% on best practices for things like this but i'm down to look into it. 

i think an easy way to reorder things is to put hidden buttons that say something like "move item up one row" or "move item down one row". how that affects  sighted keyboard users, we'd have to look into it

**Comment by alainacoppa** [comment]
@annwa I agree that that example was challenging. I was hoping that our task would a lot easier because it's just select + up/down. Hidden buttons are an interesting idea too.
